paper.tex

```diff
@@ -476,137 +476,156 @@ \section*{Appendix}
-    \item We don't view these as limitations. Rather, this is what most people are doing most of the time.
-          Just now we can finally try to do this part w/fewer errors.
-\end{enumerate}
-
-
-\aprcomment{If selection is placed on top of the demographic model, this implies
-that we are living in a soft selection world. Though sizes could just as easily
-be interpreted as carrying capacities in a hard selection model.}
-\krtcomment{We are firmly in the soft selection world here.  The connection to hard selection
-requires some other strong assumptions that will sometimes be met, and sometimes not.
-For example, if the mean fitnesses are approximately equal across demes, then a hard selection
-model could be similar to a soft selection model.  Relevant literature to dig up: Felsenstein's review,
-which will lead to other stuff.}
+% \krtcomment{
+%     We need to revisit the paragraph(s) below.
+%     We do not assume individuals are exchangeable, because we do not
+% 	require that all individuals have the same fitness.
+%     See, e.g., p 60 of Wakeley for why fitness == exchangeability
+% 	and why that is required for the standard coalescent.
+% }
+% \mhcomment{The `exchangeability' assumption also breaks down under 
+% 	different reproductive modes, since the sampling of the alleles
+%     affects coalescent times. 
+% 	For example, in a highly selfing species the coalescent time for two different alleles 
+% 	within an individual is much lower than two alleles from different individuals.
+%     I've triedto rewrite the below paragraph to update and clarify
+%     the reproduction parameters in line with the discussions on Github.
+%     Finally, 'rates' imply events occur as Poisson-like 
+% 	processes over continuous time, whereas the values used here instead define the 
+% 	\emph{fraction} of reproductions of each type.
+%     Shall we rename to selfing/clonal `fractions',
+% 	or would that mess up the rest of the schema too much?
+% }
+
+Each population has an assigned selfing rate and cloning rate, where each defines
+the probability that offspring are generated from one generation to the next by either
+self-fertilisation or cloning of an individual.
+More specifically, for a given epoch within a population denote the clonal rate by
+$\sigma$ and the selfing rate by $S$.
+$S$ and $\sigma$ can take any value between zero and one and
+can sum to more than one.
+Each generation a proportion of offspring
+$\sigma$ are expected to be generated through clonal reproduction,
+while $1-\sigma$ are expected to arise through sexual reproduction.
+Within the sexually-reproduced offspring,
+$S$ are born via self-fertilisation while the rest
+have parents drawn at random from the previous generation.
+Depending on the simulation software, this random drawing of parent may occur
+either with or without replacement. When drawing occurs with replacement, a small
+amount of ``residual'' selfing is expected, so that the realized selfing probability
```

Comment:
This is the issue that we will need to hammer out in the spec.  I'm personally annoyed that I missed this in fwdpy11. Grr....

Reply:
I wrote this section as us being agnostic to this O(1/N) difference in behavior. I can think of good arguments for and against both directions, as well as for and against us remaining agnostic.

Reply:
I really like the text here. I think we should remain pragmatic, which means we might have to leave it ambiguous and give a strong nod towards current simulator behaviour. Our aim should be to unify how to specify simulations, not to fix discrepancies between simulators and any given mathematical model. "We want X, but reality intervenes to make implementation quite awkward, so simulators do Y"

Did a new issue get opened to discuss this?

Reply:
Maybe the discussion about O(1/n) residual selfing should continue here: https://github.com/popsim-consortium/demes-spec/issues/43

Reply:
I think this is the best way of clarifying the residual selfing issue. My only suggestion - which is optional - is to highlight that the 'net' selfing rate approximates to $(1-\sigma)S$ if S >> 1/N (with a reference to [Nordborg and Donnelly 1997](https://doi.org/10.1093/genetics/146.3.1185), see their Equation 3).